src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -146,13 +147,76 @@ void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
                                         rpc::GetAllJobInfoReply *reply,
                                         rpc::SendReplyCallback send_reply_callback) {
   RAY_LOG(INFO) << "Getting all job info.";
-  auto on_done = [reply, send_reply_callback](
+
+  int limit = std::numeric_limits<int>::max();
```

Comment:
A more fundamental   problem is that we need pagination for GetAll apis, otherwise, the backend still doing a lot of work.